src/main/java/com/ericsson/ei/frontend/BackEndInformationController.java

```diff
@@ -33,97 +34,150 @@
-    public ResponseEntity<String> getInstances(Model model) {
-        return new ResponseEntity<>(backEndInstancesUtils.getInstances().toString(), getHeaders(), HttpStatus.OK);
+    public ResponseEntity<String> getInstances(Model model, HttpServletRequest request) {
+        LOG.debug("Recieved request for instances.");
+        try {
+            String activeInstance = null;
+            JsonArray allAvailableInstances = backEndInstancesUtils.getBackEndsAsJsonArray();
+
+            if (request.getSession().getAttribute("backEndInstanceName") != null) {
+                activeInstance = request.getSession().getAttribute("backEndInstanceName").toString();
+            }
+
+            allAvailableInstances = setActiveInstance(allAvailableInstances, activeInstance);
+
+            return new ResponseEntity<>(allAvailableInstances.toString(), getHeaders(), HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>(
+                    "[{\"name\":\"Unable to load instances\",\"host\":\"NO HOST\",\"port\":\"NO PORT\",\"path\":\"/\"}]",
+                    getHeaders(), HttpStatus.OK);
+        }
     }
 
     @RequestMapping(value = "/switch-backend", method = RequestMethod.POST)
     public ResponseEntity<String> switchBackEndInstance(Model model, HttpServletRequest request) {
+        LOG.debug("Recieved request to switch back end.");
         try {
-            String listOfInstances = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
-            backEndInstancesUtils.setInstances(new JsonParser().parse(listOfInstances).getAsJsonArray());
-            backEndInstancesUtils.writeIntoFile();
-            for (BackEndInformation backEndInformation : backEndInstancesUtils.getInformation()) {
-                if (backEndInformation.isActive()) {
-                    backEndInstancesUtils.setBackEndProperties(backEndInformation);
-                }
-            }
-            return new ResponseEntity<>(getHeaders(), HttpStatus.MOVED_PERMANENTLY);
+            String selectedInstanceName = getSelectedInstanceName(request);
+            request.getSession().setAttribute("backEndInstanceName", selectedInstanceName);
+
+            return new ResponseEntity<>("{\"message\": \"Backend instance with name '" + selectedInstanceName + "' was selected.\"}",
+                    getHeaders(), HttpStatus.OK);
         } catch (Exception e) {
-            return new ResponseEntity<>("Internal error" + e.getMessage(), getHeaders(), HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>("Internal error" + e.getMessage(), getHeaders(),
+                    HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
     @RequestMapping(value = "/switch-backend", method = RequestMethod.DELETE)
     public ResponseEntity<String> deleteBackEndInstance(Model model, HttpServletRequest request) {
+        LOG.debug("Recieved request to delete back end.");
```

Comment:
Why is does entrypoint and function name have different meaning. Delete and Switch has different meaning/goal, if you read from a user perspective that don't know the implementation. Should be named with same meaning/purpose.

Reply:
I cannot answer that question as I did not write the code. This change is already to large to change more that is irrelevant for this minor bug. If you want that to be improved you may want to create a new story, the code you commented here has not been updated in this PR.

Reply:
Note that I do agree with you that this is wrong, just that to change this is a too large job for this bug story that already is out of proportion. The js code needs to be updated and unittests and functional tests as well and it may take more time the expected.